packages/dm-core-plugins/src/form/fields/ArrayField.tsx

```diff
@@ -22,17 +22,26 @@ export default function ArrayField(props: TArrayTemplate) {
         enumType={attribute.enumType || undefined}
         value={value}
         label={getDisplayLabel(attribute)}
-        {...props}
       />
     )
   }
 
   if (isPrimitiveType(attribute.attributeType)) {
     return (
-      <ArrayPrimitiveTemplate
-        namePath={namePath}
-        uiAttribute={uiAttribute}
-        attribute={attribute}
+      <Controller
+        name={namePath}
+        render={({ field: { value, onChange } }) => {
+          return (
+            <ArrayPrimitiveTemplate
+              namePath={namePath}
+              onChange={onChange}
+              value={value ?? []}
+              uiAttribute={uiAttribute}
+              attribute={attribute}
+              type={attribute.attributeType}
```

Comment:
Do we need to pass type, since this already exists in the attribute prop that are passed.

Reply:
Nope, good point